src/main/scala/cromwell/engine/package.scala

```diff
@@ -14,15 +14,27 @@ import cromwell.binding.values.WdlValue
  *
  * Internally, this package is built on top of [[cromwell.binding]].
  */
-
 package object engine {
-  type WorkflowId = UUID
+
+  case class CallID(id: UUID){
+    override def toString = id.toString
+  }
+  case class WorkflowID(id: UUID) {
+    override def toString = id.toString
+  }
+  object WorkflowID {
+    def apply(id:String): WorkflowID = WorkflowID(UUID.fromString(id))
```

Comment:
The original author and another reviewer to may possibly override for this PR, but I'd like to see:

- All instances of `WorkflowID(UUID.fromString(id))` replaced with `WorkflowID(id)` using the `apply` on this line.
- All instances of `WorkflowID(UUID.randomUUID())` using a new `WorkflowID.randomID()`